src/components/single-line-diagram.js

```diff
@@ -16,14 +16,20 @@ import {makeStyles} from "@material-ui/core/styles";
 import Typography from '@material-ui/core/Typography';
 
 const useStyles = makeStyles(theme => ({
+    div: {
+        overflowX: 'auto',
+        overflowY: 'auto'
+    },
     diagram: {
-        width: 800,
-        height: 600,
+        maxWidth: 800,
```

Comment:
Ideally to be responsive it should depend on screen size. But probably we should do it in another issue.